src/ir_def/check_ir.ml

```diff
@@ -489,8 +489,9 @@ let rec check_exp env (exp:Ir.exp) : unit =
       typ exp2 <: T.nat;
       T.bool <: t
     | SameReference, [exp1; exp2] ->
-      (*FIXME typ exp1 <: T.nat;
-      typ exp2 <: T.nat;*)
+      let t1, t2 = typ exp1, typ exp2 in
+      t1 <: t2;
```

Comment:
One could put the type in `SameReference` as an argument, and test both expressions against it here. Suggestion by @crusso.

Reply:
Yeah, I think I really would.

Reply:
I do it with the other primitive. SMOP.